src/main.rs

```diff
@@ -109,42 +104,52 @@ fn dropfile_filename(filename: &str) -> String {
 }
 
 fn write_new_door32sys_dropfile(
-    orig_contents: &str, out_path: &Path, socket_fd: i64) -> Result<String, String>
-{
+    orig_contents: &str,
+    out_path: &Path,
+    socket_fd: u64,
+) -> Result<String, Error> {
     if !out_path.is_dir() {
-        return Err(format!("{} is not a directory", out_path.to_string_lossy()));
+        return Err(Error::new(
+            ErrorKind::NotFound,
+            format!("{} is not a directory", out_path.to_string_lossy()),
+        ));
     }
 
-    let dropfile_path = out_path.join(dropfile_filename(DOOR32_SYS_FILENAME));
-
     //
     //  First two lines are as follows:
     //  1 - Comm type (2=telnet)
     //  2 - Shared socket fd
     //  ...the rest is just copied over from the original.
-    //      
-    let mut contents = format!("2\r\n{}\r\n", socket_fd);
-    let remaining_lines = orig_contents.lines().skip(2);
-    for line in remaining_lines {
-        contents.push_str(&format!("{}\r\n", line));
+    //
+    let dropfile_path = out_path.join(dropfile_filename(DOOR32_SYS_FILENAME));
+    let mut file = fs::File::create(&dropfile_path)?;
+    file.write_all(b"2\r\n")?;
+    file.write_all(socket_fd.to_string().as_bytes())?;
+    file.write_all(b"\r\n")?;
+    for line in orig_contents.lines().skip(2) {
+        let cp437 = &line.to_string().into_cp437(&CP437_CONTROL).map_err(|e| {
+            Error::new(
+                ErrorKind::InvalidData,
+                format!("Failed to convert to CP437: {}", e.into_string()),
+            )
+        })?;
+        file.write_all(cp437)?;
+        file.write_all(b"\r\n")?;
     }
-
-    match contents.to_string().into_cp437(&CP437_CONTROL) {
-        Ok(cp437)   => {
-            match fs::write(&dropfile_path, cp437) {
-                Ok(()) => {
-                    println!("Created new dropfile at {}", dropfile_path.to_string_lossy());
-                    Ok(dropfile_path.to_string_lossy().to_string())
-                },
-                Err(e) => Err(e.to_string()),
-            }
-        },
-        Err(e)      => Err(format!("Failed to convert {} to CP437", e.into_string())),
-    }    
+    let pathname = dropfile_path.to_string_lossy();
+    println!("Created new dropfile at {}", &pathname);
+    Ok(pathname.to_string())
 }
 
 #[cfg(windows)]
-fn get_socket_fd(stream: TcpStream) -> Result<i64, String> {
+fn get_socket_fd(stream: TcpStream) -> Result<u64, String> {
+    use std::os::windows::io::AsRawSocket;
```

Comment:
Awesome, I didn't know I could do this. I keep liking this language more. Stuff like this makes it easy to come from C++ :)

Reply:
Yes; `use` statements can appear almost anywhere, which is nice because you can limit their scope. It's sort of like `using` from C++.